iOS_SDK/OneSignalSDK/Source/OneSignal.m

```diff
@@ -808,6 +806,12 @@ + (void)downloadIOSParamsWithAppId:(NSString *)appId {
 
         if (result[IOS_RECEIVE_RECEIPTS_ENABLE] != (id)[NSNull null])
             [OneSignalUserDefaults.initShared saveBoolForKey:OSUD_RECEIVE_RECEIPTS_ENABLED withValue:[result[IOS_RECEIVE_RECEIPTS_ENABLE] boolValue]];
+        
+        if (result[IOS_LOCATION_SHARED])
```

Comment:
This doesn't match the logic we have for Android. We need to check for key presence here.
See this commit on Android:
https://github.com/OneSignal/OneSignal-Android-SDK/pull/1124/files#diff-8f1d87ed84bc5cc2e615dcfb826df87bR2522

And this comment for the remote param override logic we want for fallback support.
https://github.com/OneSignal/OneSignal-Android-SDK/pull/1124#pullrequestreview-477112796

Reply:
Added matched logic with Android, we will need the delayed task controller to match 100% the logic, on iOS we are not waiting for remote params due to this missing behaviour